app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java

```diff
@@ -286,33 +287,32 @@ public boolean onCreateOptionsMenu(Menu menu) {
         MenuInflater inflater = getMenuInflater();
         inflater.inflate(R.menu.contribution_activity_notification_menu, menu);
 
-        if (!isThereUnreadNotifications) {
-            // TODO: used vectors are not compatible with API 19 and below, change them
-            menu.findItem(R.id.notifications).setIcon(ContextCompat.getDrawable(this, R.drawable.ic_notification_white_clip_art));
-        } else {
-            final View notification=menu.findItem(R.id.notifications).getActionView();
-            txtViewCount=notification.findViewById(R.id.notification_count_badge);
-            setNotificationCount();
-        }
-
+        notificationsMenuItem = menu.findItem(R.id.notifications);
         this.menu = menu;
-
         updateMenuItem();
-
+        setNotificationCount();
         return true;
     }
 
+    @SuppressLint("CheckResult")
     private void setNotificationCount() {
         Observable.fromCallable(() -> notificationController.getNotifications())
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(notificationList -> {
-                    Collections.reverse(notificationList);
-                    txtViewCount.setText(String.valueOf(notificationList.size()));
-                    Timber.d("Number of notifications is %d", notificationList.size());
-                }, throwable -> {
-                    Timber.e(throwable, "Error occurred while loading notifications");
-                });
+                .subscribe(this::initNotificationViews,
+                        throwable -> Timber.e(throwable, "Error occurred while loading notifications"));
+    }
+
+    private void initNotificationViews(List<Notification> notificationList) {
+        Timber.d("Number of notifications is %d", notificationList.size());
+        final View notification = notificationsMenuItem.getActionView();
+        TextView notificationCount = notification.findViewById(R.id.notification_count_badge);
```

Comment:
@maskaravivek I am still getting a NULL pointer exception here could you please help :)